windows-msvc: localtime_s/gmtime_s are header-inline CRT functions, fail to link (LNK2019) on VS 18 / MSVC 14.5x

## Summary

`libc::localtime_s` and `libc::gmtime_s` for Windows are declared as plain externs (no `link_name`), e.g. in `src/windows/mod.rs`:

```rust
pub fn localtime_s(tmDest: *mut tm, sourceTime: *const time_t) -> errno_t;
```

In the UCRT these names are `static __inline` header wrappers (in `time.h`) that forward to `_localtime64_s`/`_gmtime64_s`; only the underscore-prefixed variants are actual exports of ucrtbase / `api-ms-win-crt-time-l1-1-0`. Microsoft's docs state localtime_s is implemented as an inline function.

## Behavior change

Historically this linked anyway with VS 2022-era import libraries. With Visual Studio 18 toolchains (MSVC 14.50 / 14.51 — which GitHub's `windows-latest` image now ships), linking any program that calls `libc::localtime_s`/`gmtime_s` fails:

```
error LNK2019: unresolved external symbol localtime_s
error LNK2019: unresolved external symbol gmtime_s
```

Same code, same libc version, links fine on VS 2022 (14.4x) — the compat objects that used to satisfy these names appear to be gone from the newer import libs.

## Repro

On `x86_64-pc-windows-msvc` with a VS 18 toolchain:

```rust
fn main() {
    unsafe {
        let t: libc::time_t = 0;
        let mut tm: libc::tm = core::mem::zeroed();
        libc::localtime_s(&mut tm, &t);
    }
}
```

## Suggested fix

Declare them with the exported symbol names, matching what the header inline does (libc's Windows `time_t` is `i64`, so the 64-bit variants are the right targets):

```rust
#[link_name = "_localtime64_s"]
pub fn localtime_s(tmDest: *mut tm, sourceTime: *const time_t) -> errno_t;
#[link_name = "_gmtime64_s"]
pub fn gmtime_s(tmDest: *mut tm, sourceTime: *const time_t) -> errno_t;
```

Other UCRT declarations that exist only as header inlines may deserve the same audit.

## Downstream reports

- https://github.com/ianm199/omnilua/issues/308 (end users on VS 18 unable to `cargo install`; five releases of CI on VS 2022 runners linked fine)
- https://github.com/wowemulation-dev/rilua/issues/21

--- a/src/windows/mod.rs
+++ b/src/windows/mod.rs
@@ -419,8 +419,11 @@ extern "C" {
         } else {
             pub fn ctime(sourceTime: *const time_t) -> *mut c_char;
             pub fn difftime(timeEnd: time_t, timeStart: time_t) -> c_double;
+            #[link_name = "_gmtime64_s"]
             pub fn gmtime_s(destTime: *mut tm, srcTime: *const time_t) -> c_int;
+            #[link_name = "_localtime64_s"]
             pub fn localtime_s(tmDest: *mut tm, sourceTime: *const time_t) -> crate::errno_t;
+            #[link_name = "_time64"]
             pub fn time(destTime: *mut time_t) -> time_t;
         }
     }

